server/paststream_poller.js

```diff
@@ -0,0 +1,47 @@
+function createPollRoute(channelID) {
+    return `https://holodex.net/api/v2/channels/${channelID}/videos?lang=en&type=stream%2Cplaceholder&include=live_info&limit=24&offset=0&paginated=true`
+}
+
+async function fetchPaststreamPage(channelID) {
+    try {
+        const res = await fetch(createPollRoute(channelID))
+        if (res.status !== 200) {
+            return { error: `HTTP status: ${res.status}`, result: null }
+        }
+        const youtubeJSON = await res.json()
+        return { error: null, result: youtubeJSON }
+    } catch (e) {
+        return { error: e.toString(), result: { items: [] } }
+    }
+}
+
+function extractPaststreamInfo(fromPageContent) {
+    const lastStream = fromPageContent.items.find(e => {
+        return e.status === 'past' &&
+            e.type === 'stream' &&
+            e.topic_id !== 'shorts' &&
+            e.duration >= 1800
+    }) || null
+    return lastStream ? {
+        // Whitelisting the fields to make frontend debugging easier
+        endActual: Date.parse(lastStream.end_actual)
+    } : null
+}
+
+export async function pollPaststreamStatus(channelID) {
+    if (process.env.USE_DUMMY_DATA === "true") {
+        return pollPaststreamStatusDummy(process.env.WATCH_CHANNEL_ID)
+	}
+
+    const { error, result: youtubeJSON } = await fetchPaststreamPage(channelID)
+    if (error) {
+        return { error, result: null }
+    }
+
+    return extractPaststreamInfo(youtubeJSON)
```

Comment:
Please wrap this in a `{ error, result }` object to be consistent with the error case.